packages/sdk-cli/src/models/localRelay/index.ts

```diff
@@ -29,9 +29,28 @@ export async function instance(): Promise<RelayInfo> {
   }
 
   const relayJsPath = await relayEntryPointPath();
-  const logStream = createWriteStream(RELAY_LOG_FILE_PATH);
 
-  launch([relayJsPath], logStream);
+  createWriteStream(RELAY_LOG_FILE_PATH)
+    .on('open', (fd) => {
+      const relayProcess = launch([relayJsPath], fd);
+
+      relayProcess.on('error', (error) => {
+        console.error('Local relay process threw error');
+        relayProcess.kill('SIGKILL');
+        throw error;
+      });
+
+      relayProcess.on('close', async () => {
+        console.log('Local relay process exited and closed');
+      });
+    })
+    .on('error', (error) => {
+      console.error(
+        `Error creating an output stream to file at path: ${RELAY_LOG_FILE_PATH}`,
+      );
+      throw error;
```

Comment:
What does this actually do, re-throwing? You're already in an event handler, is this not just going to create an unhandled error? Or is that the point?

Reply:
Perhaps turn `createWriteStream(RELAY_LOG_FILE_PATH)` into a Promise:

```
const fd = await new Promise((resolve, reject) => {
  createWriteStream(RELAY_LOG_FILE_PATH)
    .on('open', resolve)
    .on('error', reject);
});
```

Reply:
Tried to do something similar with the `subprocess` too, but too much hassle. Left it as is

Reply:
https://github.com/Fitbit/developer-bridge/pull/176/commits/c8b70fddcb1921f878d20f7e25c27e8d6c141005